crates/services/sync/src/import/tests.rs

```diff
@@ -1026,6 +1027,65 @@ async fn import__missing_transactions_sends_peer_report() {
         .await;
 }
 
+#[tokio::test]
+async fn import__execution_error_on_header_4_when_awaits_for_1000000_blocks() {
```

Comment:
I'm not sure I understand this test. What does `1000000` have to do with the error? Is it just to show that we don't import them out of order? i.e. if header 4 doesn't import correctly, none after will either?

Reply:
It shows that if the first query fails, it stops requesting all remaining queries. You can check [this commit](https://github.com/FuelLabs/fuel-core/pull/1916/commits/241e6ecf951a14ce2dbfd361e38c5a5a0e05fc6c) to see the problematic place=)

Reply:
Okay. I see. Makes sense.